includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -276,6 +278,14 @@ protected function get_row_data( $item ) {
 
 		switch ( $this->type ) {
 			case 'courses':
+				// Get Learners (i.e. those who have started)
+				$course_args     = array(
+					'post_id' => $item->ID,
+					'type'    => 'sensei_course_status',
+					'status'  => 'any',
+				);
+				$course_students = Sensei_Utils::sensei_check_for_activity( apply_filters( 'sensei_analysis_course_learners', $course_args, $item ), true );
```

Comment:
Got an error when switching to Courses.
```
10-Mar-2022 13:07:27 UTC] PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Sensei_Analysis_Overview_List_Table::get_average_progress() must be of the type array, object given, called in /Users/merkushin/Developer/sensei/includes/class-sensei-analysis-overview-list-table.php on line 336 and defined in /Users/merkushin/Developer/sensei/includes/class-sensei-analysis-overview-list-table.php:523
Stack trace:
#0 /Users/merkushin/Developer/sensei/includes/class-sensei-analysis-overview-list-table.php(336): Sensei_Analysis_Overview_List_Table->get_average_progress(138, Object(WP_Comment))
#1 /Users/merkushin/Developer/sensei/includes/class-sensei-list-table.php(230): Sensei_Analysis_Overview_List_Table->get_row_data(Object(WP_Post))
#2 /Users/merkushin/Local Sites/devwp/app/public/wp-admin/includes/class-wp-list-table.php(1377): Sensei_List_Table->single_row(Object(WP_Post))
#3 /Users/merkushin/Local Sites/devwp/app/public/wp-admin/includes/class-wp-list-table.php(1362): WP_List_Table->display_rows()
#4 /Users/merkushin/Local Sites/devwp/app/public/wp-admin/includes/class-wp-list-table.php(1289): WP_List_Table->display_rows_or_placeholder()
#5 /Users/merkushin/Developer/sensei/includes/class-sensei-analysis.php(194): WP_List_Table->display()
#6 /Users/merkushin/Developer/sensei/includes/class-sensei-analysis.php(170): Sensei_Analysis->analysis_default_view('courses')
#7 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(307): Sensei_Analysis->analysis_page('')
#8 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
#9 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/plugin.php(474): WP_Hook->do_action(Array)
#10 /Users/merkushin/Local Sites/devwp/app/public/wp-admin/admin.php(259): do_action('course_page_sen...')
#11 /Users/merkushin/Local Sites/devwp/app/public/wp-admin/edit.php(10): require_once('/Users/merkushi...')
#12 {main}
  thrown in /Users/merkushin/Developer/sensei/includes/class-sensei-analysis-overview-list-table.php on line 523
```
<img width="861" alt="CleanShot 2022-03-10 at 16 09 01@2x" src="https://user-images.githubusercontent.com/329356/157668412-bc53f351-c23e-47a0-aca7-ca8c949752b6.png">

Here is the reason for that: https://github.com/Automattic/sensei/blob/update/add_average_progress_courses/includes/class-sensei-utils.php#L148-L148

I came across this behavior earlier. For me, it is pretty unexpected, but...

The common workaround is something like that:
```
$course_students = Sensei_Utils::sensei_check_for_activity( apply_filters( 'sensei_analysis_course_learners', $course_args, $item ), true );
if ( ! is_array( $course_students ) ) {
	$course_students = array( $course_students );
}
```